src/app/middleware.ts

```diff
@@ -0,0 +1,24 @@
+// middleware.ts
+
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+
+import dbConnect from '@/utils/mongodb';
+
+export async function middleware(request: NextRequest) {
+  if (request.nextUrl.pathname.startsWith('/api')) {
+    try {
+      await dbConnect();
+    } catch (error) {
+      console.error('Database connection error:', error);
```

Comment:
remove console errors

Reply:
resolved in [f5e7758](https://github.com/anasbinarif/ace-my-exam/pull/10/commits/f5e7758c9a2c68e71e1d45f4114d9629a3a87510)